tests/integration/test_application.py

```diff
@@ -331,3 +333,14 @@ async def test_app_charm_name():
         await model.wait_for_idle(status="active")
         assert 'ubuntu' in app.charm_url
         assert 'ubuntu' == app.charm_name
+
+
+@base.bootstrapped
+async def test_app_relation_destroy_block_until_done():
+    async with base.CleanModel() as model:
+        app: Application = await model.deploy('docker-registry')
+        rsa: Application = await model.deploy("easyrsa")
+        await app.relate("cert-provider", rsa.name)
+        await model.wait_for_idle(status="active")
+        await app.destroy_relation("cert-provider", rsa.name, block_until_done=True)
+        await app.relate("cert-provider", rsa.name)
```

Comment:
Instead of relating again, let's call the `relations` here and make sure we don't see the relation there.

Reply:
Done, thanks for the feedback!